api/apps/query-builder/pdf/post.ts

```diff
@@ -0,0 +1,39 @@
+import createAPIGatewayProxyHandler from "package/backend/createAPIGatewayProxyHandler";
+import { createZip } from "./Pdf";
+
+type Params = {
+  files: [
+    {
+      title: string;
+      content: string;
+    }
+  ];
+};
+
+export const logic = async ({ files }: Params) => {
+  try {
+    const zip = await createZip(files);
+    const zipBuffer = await zip.generateAsync({ type: "nodebuffer" });
+    const zipString = zipBuffer.toString("base64");
+    if (!zipString) throw new Error("Failed to create zip");
```

Comment:
as I mentioned in the other PR, I could see us instead uploading to S3 and providing a download link